connect/runtime/src/test/java/org/apache/kafka/connect/runtime/rest/RestServerTest.java

```diff
@@ -166,19 +173,21 @@ public void testOptionsDoesNotIncludeWadlOutput() throws IOException {
 
         HttpOptions request = new HttpOptions("/connectors");
         request.addHeader("Content-Type", MediaType.WILDCARD);
-        CloseableHttpClient httpClient = HttpClients.createMinimal();
-        HttpHost httpHost = new HttpHost(
-            server.advertisedUrl().getHost(),
-            server.advertisedUrl().getPort()
-        );
-        CloseableHttpResponse response = httpClient.execute(httpHost, request);
-        Assert.assertEquals(MediaType.TEXT_PLAIN, response.getEntity().getContentType().getValue());
-        ByteArrayOutputStream baos = new ByteArrayOutputStream();
-        response.getEntity().writeTo(baos);
-        Assert.assertArrayEquals(
-            request.getAllowedMethods(response).toArray(),
-            new String(baos.toByteArray(), StandardCharsets.UTF_8).split(", ")
-        );
+        try (CloseableHttpClient httpClient = HttpClients.createMinimal()) {
```

Comment:
I can understand why we went back to try-with-resources for responses, since we create more than one of them in some tests. But we only ever create one client in our tests, and it's always `HttpClients.createMinimal()`. I think it'd be cleaner if we went back to a single class-level `httpClient` field that gets closed automatically during `tearDown`. We could even unconditionally initialize it in `setUp` to save the trouble of repeating that line for every test case that uses a client.

Reply:
Right, went a bit too far in the try-with-resources implementation.